RIOT-OS/RIOT
# riot.doxyfile: Update and remove deprecated options

### Contribution description
While testing https://github.com/RIOT-OS/RIOT/pull/20896, I noticed that my somewhat recent Doxygen (up-to-date Arch Linux, doxygen version 1.12.0) is complaining about some obsolete options:

```
warning: Tag 'HTML_TIMESTAMP' at line 1197 of file '-' has become obsolete.
         To avoid this warning please remove this line from your configuration file or upgrade it using "doxygen -u"
warning: Tag 'FORMULA_TRANSPARENT' at line 1478 of file '-' has become obsolete.
         To avoid this warning please remove this line from your configuration file or upgrade it using "doxygen -u"
warning: Tag 'LATEX_SOURCE_CODE' at line 1752 of file '-' has become obsolete.
         To avoid this warning please remove this line from your configuration file or upgrade it using "doxygen -u"
warning: Tag 'DOCBOOK_PROGRAMLISTING' at line 1914 of file '-' has become obsolete.
         To avoid this warning please remove this line from your configuration file or upgrade it using "doxygen -u"
warning: Tag 'CLASS_DIAGRAMS' at line 2123 of file '-' has become obsolete.
         To avoid this warning please remove this line from your configuration file or upgrade it using "doxygen -u"
warning: Tag 'DOT_FONTNAME' at line 2165 of file '-' has become obsolete.
         To avoid this warning please remove this line from your configuration file or upgrade it using "doxygen -u"
warning: Tag 'DOT_FONTSIZE' at line 2172 of file '-' has become obsolete.
         To avoid this warning please remove this line from your configuration file or upgrade it using "doxygen -u"
warning: Tag 'DOT_TRANSPARENT' at line 2381 of file '-' has become obsolete.
         To avoid this warning please remove this line from your configuration file or upgrade it using "doxygen -u"
```

So I did what the output told me to and ran `doxygen -u doc/doxygen/riot.doxyfile`. The result is this PR.



### Testing procedure
The above warnings should not show up in a doxygen build anymore, the output of the HTML should not change.



### Issues/PRs references
Noticed in #20896, but unrelated.
